assets/src/components/parts/janus-slider/Slider.tsx

```diff
@@ -0,0 +1,141 @@
+/* eslint-disable react/prop-types */
+import React, { CSSProperties, useEffect, useState } from 'react';
+import { parseBoolean } from 'utils/common';
+
+// TODO: fix typing
+const Slider: React.FC<any> = (props) => {
+  const [state, setState] = useState<any[]>(Array.isArray(props.state) ? props.state : []);
+  const [model, setModel] = useState<any>(Array.isArray(props.model) ? props.model : {});
+  const id: string = props.id;
+  useEffect(() => {
+    if (typeof props?.model === 'string') {
+      setModel(JSON.parse(props.model));
+    }
+    if (typeof props?.state === 'string') {
+      setState(JSON.parse(props.state));
+    }
+  }, [props]);
+
+  const {
+    x,
+    y,
+    z,
+    width,
+    height,
+    src,
+    alt,
+    customCssClass,
+    label,
+    maximum,
+    minimum,
+    snapInterval,
+    showDataTip,
+    showValueLabels,
+    showThumbByDefault,
+    showLabel,
+    showTicks,
+    invertScale,
+    value,
+  } = model;
+  const styles: CSSProperties = {
+    position: 'absolute',
+    width: `${width}px`,
+    top: `${y}px`,
+    left: `${x}px`,
+    height: `${height}px`,
+    zIndex: z,
+    flexDirection: model.showLabel ? 'column' : 'row',
+  };
+  const inputStyles: CSSProperties = {
+    width: '100%',
+    height: `${height}px`,
+    zIndex: z,
+    direction: invertScale ? 'rtl' : 'ltr',
+  };
+  const divStyles: CSSProperties = {
+    width: '100%',
+    display: `flex`,
+    flexDirection: 'row',
+  };
+
+  const [sliderValue, setSliderValue] = useState(value ? value : minimum); // if value is undefined, set default to minimum value;
+  const [isSliderEnabled, setIsSliderEnabled] = useState(true);
+
+  const inputWidth: any = document.getElementById(`${id}`)?.getBoundingClientRect().width;
+  const thumbWidth: any = document.getElementById('slider-thumb')?.getBoundingClientRect().width;
```

Comment:
With the way that this coded, there will be multiple DOM elements that get assigned the id of "slider-thumb" if there are multiple instances of the `Slider` component present in the same view.   In the case where there is more than one DOM element with the same id, a call to `document.getElementById` will return a reference to the *first* element with that identifier.  Is that going to work here?  Can the widths of these 'slider-thumb' components ever differ?  

If this isn't a concern due to invariants that either the widths never differ across instances, or there are never multiple instances, this should be documented as such.  Otherwise, the fix here would be to dynamically set the id.